platzky/attachment/mime_validation.py

```
@@ -0,0 +1,151 @@
+        return
+
+    # Use puremagic to detect the actual content type
+    try:
+        detected = puremagic.magic_string(content)
+    except puremagic.PureError:
+        if allow_unrecognized:
+            return
+        raise ContentMismatchError(
+            f"Content of '{filename}' does not match declared MIME type '{mime_type}'. "
+            "Could not identify file type from content."
+        )
```

Ruff (0.14.13)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)